src/main/java/com/entropyteam/entropay/employees/services/ContractService.java

```diff
@@ -28,14 +33,45 @@ public class ContractService extends BaseService<Contract, ContractDto, UUID> {
 
     @Autowired
     public ContractService(ContractRepository contractRepository, CompanyRepository companyRepository,
-            EmployeeRepository employeeRepository, RoleRepository roleRepository,
-            SeniorityRepository seniorityRepository) {
+                           EmployeeRepository employeeRepository, RoleRepository roleRepository,
+                           SeniorityRepository seniorityRepository) {
         this.contractRepository = contractRepository;
         this.companyRepository = companyRepository;
         this.employeeRepository = employeeRepository;
         this.roleRepository = roleRepository;
         this.seniorityRepository = seniorityRepository;
+    }
+
+    @Transactional
+    @Override
+    public ContractDto create(ContractDto contractDto) {
+        contractRepository.findContractByEmployeeAndActiveIsTrue(
+                        employeeRepository.findById(contractDto.employeeId()).orElseThrow()
+                )
+                .ifPresent(existent -> {
+                    existent.setActive(false);
+                    contractRepository.saveAndFlush(existent);
+                });
+        return super.create(contractDto);
+    }
+
+    @Transactional
+    public ContractDto modifyStatus(UUID contractId, boolean setActive) {
+        Contract contract = contractRepository.findById(contractId)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Entity not found"));
+
+        if ((!setActive && !contract.isActive()) || (setActive && contract.isActive()))
+            throw new ResponseStatusException(HttpStatus.NO_CONTENT, "Contract " + contract.getId() + " already " + (setActive ? "active" : "inactive")
+                    + " for employee " + contract.getEmployee().getId());
 
+        if (setActive) {
+            contractRepository.findContractByEmployeeAndActiveIsTrue(contract.getEmployee()).ifPresent(existent -> {
+                existent.setActive(false);
+                contractRepository.save(existent);
+            });
+        }
+        contract.setActive(setActive);
+        return toDTO(contractRepository.save(contract));
     }
```

Comment:
I would refactor this method to receive only the contract Id, activate the contract and deactivate the rest of them. Maybe we could change the method name also.

Reply:
Same as before, we also need to have the ability to deactivate a contract, according to the ticket.